src/encryption-key.ts

```diff
@@ -0,0 +1,51 @@
+import { binaryStringToBytes, bytesToBinaryString, decodeSafe64, encodeSafe64 } from './util';
+
+/**
+ * An key that can be used to encrypt and decrypt data
+ * This wrapper ensures that keys can be safely serialized as JSON (by encoding them as URL-Safe Base64)
+ * and avoids confusion when dealing with converting to and from this encoded format.
+ */
+export class EncryptionKey {
+  /**
+   * Create an {@link EncryptionKey} from encoded URL-safe base 64 version of the key
+   */
+  public static fromSerialized(value: string) {
+    return new EncryptionKey(binaryStringToBytes(decodeSafe64(value || '')));
+  }
+  /**
+   * @deprecated
+   * Create an {@link EncryptionKey} from a binary string version of the key
+   */
+  public static fromRaw(value: string) {
+    return new EncryptionKey(binaryStringToBytes(value));
+  }
+  public static fromBytes(bytes: Uint8Array) {
+    return bytesToBinaryString(bytes);
```

Comment:
This should return an instance of `EncryptionKey`

Reply:
done